sys/riscv/include/atomic.h

```diff
@@ -137,34 +137,45 @@ ATOMIC_FCMPSET_ACQ_REL(16);
 #define	atomic_fcmpset_acq_short	atomic_fcmpset_acq_16
 #define	atomic_fcmpset_rel_short	atomic_fcmpset_rel_16
 
-static __inline void
-atomic_add_32(volatile uint32_t *p, uint32_t val)
-{
-
-	(void)__atomic_add_fetch(p, val, __ATOMIC_RELAXED);
-
-}
-
-static __inline void
-atomic_subtract_32(volatile uint32_t *p, uint32_t val)
-{
-
-	(void)__atomic_sub_fetch(p, val, __ATOMIC_RELAXED);
-}
+#define	ATOMIC_OP_IMPL(WIDTH, OP, FN, PRE)					\
+	static __inline void							\
+	atomic_##OP##_##WIDTH(volatile uint##WIDTH##_t *p, uint##WIDTH##_t val)	\
+	{									\
+		PRE;								\
+		(void)__atomic_##FN##_fetch(p, val, __ATOMIC_RELAXED);		\
+	}
```

Comment:
Do we not need acq/rel?

Reply:
They appear to be unused but we should provide them